--- a/configure
+++ b/configure
@@ -4004,12 +4004,12 @@ ALL_LINGUAS="cs de es eu fi fr nl ru zh_
 # Checks for libraries.
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for wbkgdset in -lcurses" >&5
-$as_echo_n "checking for wbkgdset in -lcurses... " >&6; }
+$as_echo_n "checking for wbkgdset in -lncursesw... " >&6; }
 if test "${ac_cv_lib_curses_wbkgdset+set}" = set; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lcurses  $LIBS"
+LIBS="-lncursesw  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -4056,7 +4056,7 @@ if test "${ac_cv_lib_ncurses_wbkgdset+se
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lncurses  $LIBS"
+LIBS="-lncursesw  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -4523,9 +4523,9 @@ fi
 
 # We prefer ncurses to curses if both are found.
 if test -n "$HAVE_NCURSES_H" -a -n "$HAVE_LIBNCURSES";  then
-   LIBS="-lncurses $LIBS"
+   LIBS="-lncursesw $LIBS"
 elif test -n "$HAVE_CURSES_H" -a -n "$HAVE_LIBCURSES"; then
-   LIBS="-lcurses $LIBS"
+   LIBS="-lncursesw $LIBS"
 else
    echo -e "Error:  both library and header files for either curses or ncurses"\
    	"library\nis required to build this package.  See INSTALL file for"\
