--- a/unix/Makefile.in
+++ b/unix/Makefile.in
@@ -250,7 +250,7 @@ MAN_FLAGS               = @MAN_FLAGS@
 
 # If non-empty, install the timezone files that are included with Tcl,
 # otherwise use the ones that ship with the OS.
-INSTALL_TZDATA		= @INSTALL_TZDATA@
+INSTALL_TZDATA		=
 
 #--------------------------------------------------------------------------
 # The information below is usually usable as is. The configure script won't
@@ -768,7 +768,7 @@ trace-test: ${TCLTEST_EXE}
 # Installation rules
 #--------------------------------------------------------------------------
 
-INSTALL_BASE_TARGETS = install-binaries install-libraries install-msgs $(INSTALL_TZDATA)
+INSTALL_BASE_TARGETS = install-binaries install-libraries
 INSTALL_DOC_TARGETS = install-doc
 INSTALL_PACKAGE_TARGETS = install-packages
 INSTALL_DEV_TARGETS = install-headers
